lib/cisco_node_utils/fabricpath_global.rb

```diff
@@ -142,15 +139,15 @@ def default_allocate_delay
       config_get_default('fabricpath', 'allocate_delay')
     end
 
-    def graceful_merge
-      graceful_merge_conf = config_get('fabricpath', 'graceful_merge')
-      return default_graceful_merge if graceful_merge_conf.nil?
-      graceful_merge_conf.first == 'disable' ? false : true
+    def auto_switch_id
+      config_get('fabricpath', 'auto_switch_id')
     end
 
-    def auto_switch_id
-      val = config_get('fabricpath', 'auto_switch_id')
-      val.first.to_i
+    def graceful_merge
+      graceful_merge_conf = config_get('fabricpath', 'graceful_merge')
+      # opposite meaning with the cli
+      return true if graceful_merge_conf.nil?
+      graceful_merge_conf == true ? false : true
```

Comment:
Ok agree

From: Mike Wiebe <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-node-utils <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Friday, November 13, 2015 at 8:38 AM
To: cisco/cisco-network-node-utils <cisco-network-node-utils@noreply.github.com<mailto:cisco-network-node-utils@noreply.github.com>>
Cc: Cisco Employee <dcherian@cisco.com<mailto:dcherian@cisco.com>>
Subject: Re: [cisco-network-node-utils] adding fabricpath topo and fixed feature_set as a class method (#75)


In lib/cisco_node_utils/fabricpath_global.rb<https://github.com/cisco/cisco-network-node-utils/pull/75#discussion_r44803307>:

>      end
>
> -    def auto_switch_id
> -      val = config_get('fabricpath', 'auto_switch_id')
> -      val.first.to_i
> +    def graceful_merge
> +      graceful_merge_conf = config_get('fabricpath', 'graceful_merge')
> +      # opposite meaning with the cli
> +      return true if graceful_merge_conf.nil?
> +      graceful_merge_conf == true ? false : true


Suggest: graceful_merge_conf ? false : true instead.

-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-node-utils/pull/75/files#r44803307>.